### PR TITLE
verify that the context can be read

### PR DESCRIPTION
**Pull request by l8nightcaffeine**

### CODE REVIEW

File: Nancy.Rollbar/RollbarDataFactory.cs [hunk 1]
```diff
@@ -78,8 +78,10 @@ public class RollbarDataFactory : IRollbarDataFactory {
                     .ToDictionary(x => x.Key, x => x.Value);
                 if (rr.PostParams.Count == 0) {
                     rr.PostParams = null;
-                }
-                rr.PostBody = context.Request.Body.AsString();
+                }
+                if (!context.Request.Body.CanRead) {
+                    rr.PostBody = context.Request.Body.AsString();
```
Comment (aelnagger) [inline Nancy.Rollbar/RollbarDataFactory.cs:83]: Are we going to have an issue if we can't read and we leave the PostBody null?

Comment (l8nightcaffeine) [inline Nancy.Rollbar/RollbarDataFactory.cs:83]: only if you don't check if its empty or null. (The only time you should be modifying the payload is in the scrubber, which isn't implemented in this library)

We could fill it with an empty string....but I feel as though leaving it as null signifies all did not go as planned.

File: Nancy.Rollbar/RollbarDataFactory.cs [hunk 2]
```diff
@@ -78,8 +78,10 @@ public class RollbarDataFactory : IRollbarDataFactory {
                     .ToDictionary(x => x.Key, x => x.Value);
                 if (rr.PostParams.Count == 0) {
                     rr.PostParams = null;
-                }
-                rr.PostBody = context.Request.Body.AsString();
+                }
+                if (!context.Request.Body.CanRead) {
```
Comment (aelnagger) [inline Nancy.Rollbar/RollbarDataFactory.cs:82]: Also, you definitely want to flip this condition.

**Comment by l8nightcaffeine** [comment]
CanRead does not tell you if the stream is still open, and the using statement within .AsString() is closing the stream. So if you have multiple errors within the same context, the Stream will be closed.

File: Nancy.Rollbar/RollbarDataFactory.cs (hunk 1 above)
Comment (Crisfole) [inline Nancy.Rollbar/RollbarDataFactory.cs:83]: The JsonSerializer won't send the key if PostBody is null. So there's actually a semantic difference between the two.

**Pull request closed without merging**